experimental/python/wheel.bzl

```diff
@@ -140,8 +140,14 @@ def _py_wheel_impl(ctx):
         for r in requirements:
             args.add("--extra_requires", r + ";" + option)
 
+    for group, entries in ctx.attr.entry_points.items():
+        for e in entries:
+            # Assemble a string that can be passed as a CLI argument and parsed unambiguously.
+            args.add("--entry_point", group + ";" + e)
+
+    # Merge console_scripts into entry_points
     for name, ref in ctx.attr.console_scripts.items():
-        args.add("--console_script", name + " = " + ref)
+        args.add("--entry_point", "console_scripts;{name} = {ref}".format(name = name, ref = ref))
```

Comment:
Perhaps it would be better to generate entry_points.txt in .bzl (like _target_wrapped_inputs.txt does above) and just package it in wheelmaker.py.

I leave it up to you to decide which approach you prefer.

Reply:
Rewrote entirely. The ugliness moved, but is not gone. The wheelmaker.py file is much cleaner though.